service/history/transferQueueTaskExecutorBase.go

```diff
@@ -319,7 +320,17 @@ func (t *transferQueueTaskExecutorBase) recordWorkflowClosed(
 	if archiveVisibility {
 		ctx, cancel := context.WithTimeout(context.Background(), t.config.TransferProcessorVisibilityArchivalTimeLimit())
 		defer cancel()
-		_, err := t.historyService.archivalClient.Archive(ctx, &archiver.ClientRequest{
+
+		saTypeMap, err := searchattribute.BuildTypeMap(t.config.ValidSearchAttributes)
```

Comment:
why not making this `saTypeMap` global?
i see this new variable being passed all over the place

Reply:
It comes from dynamic config and has to be evaluated every time. This might be changed in upcoming PRs though.